Source/Core/Core/IOS/IOSC.cpp

```diff
@@ -327,8 +401,22 @@ ReturnCode IOSC::VerifyPublicKeySign(const std::array<u8, 20>& sha1, Handle sign
     mbedtls_mpi_read_binary(&rsa.E, reinterpret_cast<const u8*>(&entry->misc_data), 4);
     rsa.len = entry->data.size();
 
-    const int ret = mbedtls_rsa_pkcs1_verify(&rsa, nullptr, nullptr, MBEDTLS_RSA_PUBLIC,
-                                             MBEDTLS_MD_SHA1, 0, sha1.data(), signature.data());
+    int ret = mbedtls_rsa_pkcs1_verify(&rsa, nullptr, nullptr, MBEDTLS_RSA_PUBLIC, MBEDTLS_MD_SHA1,
+                                       0, sha1.data(), signature.data());
+    if (ret != 0 && m_console_type == ConsoleType::RVT)
+    {
+      // Some dev signatures do not have proper PKCS#1 padding. Just powmod and check it ends with
+      // digest.
+      std::vector<u8> result(signature.size());
+      if (!mbedtls_rsa_public(&rsa, signature.data(), result.data()) &&
+          !memcmp(&result[result.size() - sha1.size()], sha1.data(), sha1.size()))
+      {
+        ret = 0;
+      }
+    }
+
+    mbedtls_rsa_free(&rsa);
```

Comment:
This results in a double free. There's already a scope guard to ensure mbedtls_rsa_free is called even if we return early.

```cpp
    Common::ScopeGuard context_guard{[&rsa] { mbedtls_rsa_free(&rsa); }};
```